log/log_test.go

```diff
@@ -437,4 +437,53 @@ func TestLog(t *testing.T) {
 
 		So(string(bytesWritten), ShouldResemble, "styled output\n")
 	})
+
+	Convey("destination is protected against data races", t, func() {
+
+		Convey("Given a process logging from a goroutine", func() {
+			letTheRaceBegin := make(chan bool)
+			letTheRaceEnd := make(chan bool)
+			go func() {
+				<-letTheRaceBegin
+				printEvent([]byte{1, 2, 3})
+				close(letTheRaceEnd)
+			}()
+
+			Convey("When I change the destination it should not cause a data race", func() {
+				close(letTheRaceBegin)
+				SetDestination(io.Discard, nil)
+				<-letTheRaceEnd
+				So(true, ShouldEqual, true) // all we are testing for is the absence of detecting a data race
+			})
+
+		})
+
+		Convey("Given the standard destination returns an error", func() {
+			destination = WriteWillError{}
```

Comment:
destination not used, so can be removed

Reply:
this (and the types above) are used within the log.go file so that writes go to the fallbackDestination instead

Reply:
Yep misunderstood this bit but looking back over it I can see this is to force any error on the destination so as you say the code falls back onto the "fallbackDestination" and therefore testing the race conditon on this global variable.